### DISCUSSION TITLE
Creating Automation Scripts for VPN Server/Infrastructure setup in :cloud:

### DISCUSSION THREAD

**Discussion by IranZop** [WeAreMahsaAmini/Core#249] [Ideas]
If people outside of Iran run VPN servers on random ports and share them with people in Iran, Given that these servers will receive less traffic in comparison to a well-known VPN service provider, it would be harder for the government to identify and kill all connections.   

The project below needs some TCL to run in the AWS cloud today. I'm actively working on it to get it to work:
https://github.com/IranZop/openvpn_aws_cloudformation

Once finished, anyone with an AWS account can deploy a very secure infrastructure containing an EC2 instance running OpenVPN on a random port with just a click of a button, directly from GitHub.

**Comment by sdimitt** [comment]
Have you checked to see if someone from Iran can actually access it. I believe AWS blocks IP addresses from Iran.

**Comment by IranZop** [comment]
@sdimitt @aboutsajjad  @syavash 
I deployed a test AWS Lambda and tried to hit the endpoint from multiple cloud service providers in Iran and had no issues:
`curl 'https://3jvxzxpgji.execute-api.us-east-1.amazonaws.com/dev/msgy/hello?query=test'` 

I also host multiple static websites using AWS CloudFront, AWS S3 and they work fine in Iran.

With regards to #62, we can change the EC2 Bootstrap instructions [here](https://github.com/IranZop/openvpn_aws_cloudformation/blob/master/cloudFormation/openVPNSingleInstance.yaml#L181) to setup a shadowsocks+v2ray service instead of OpenVPN.

The idea is to make this process as simple as possible so that anyone can deploy the full stack with a click of a button. Check [this website](https://noise.getoto.net/2020/10/19/event-driven-architecture-for-using-third-party-git-repositories-as-source-for-aws-codepipeline/) and look for the "Launch Stack" Button. once our project is ready to go, we can put a button similar to the one on this site in git README file and anyone can deploy the service without understanding what's happening under the hood. Given that the project source is available to everyone, experts can review the stack and make sure it's safe AND we can keep the scripts up-to-date with changes in Iran's filtering logic. 

Let me know your thoughts :)

**Reply by hiddify-com** [reply]
Please checkout https://github.com/WeAreMahsaAmini/Core/discussions/290
It uses Microsoft Azure for deploying MTProxy and TLS Proxy

**Comment by arashmhashemi** [comment]
This is an excellent idea @IranZop I've been providing OpenVPN to our fellow Iranians since Aban, so a few points here:

1. OpenVPN is the only reliable VPN that works most of the time. Almost all other protocols are blocked
2. AWS is too expensive because of costly egress, which may not make sense to an average user
3. Azure and GCP are also expensive
4. I use Hetzner. With less than $5/month, you get 20T traffic which is suitable for ~10k users based on my calculation
5. Running OpenVPN on 443 makes it accessible most of the times
6. I use an open source admin dashboard and let admins in Iran manage it, looking for volunteer admins
7. I use docker to deploy everything, so not so hard to automate if we want

If you want, let's schedule some time and discuss this further.

**Reply by 0xland** [reply]
I'm in Iran and Openvpn usually works on mild restriction hours and land internet providers, but on severe restriction hours and cellular network the only protocol works non stop without issue is method mentioned in following discussion https://github.com/WeAreMahsaAmini/Core/discussions/62

**Reply by poorp** [reply]
Is it still accessible on port 443? What config are you providing? Can I get a short term test conf from you and discuss this more in private?
I'm trying to get something reliable up and running.

**Comment by hiddify-com** [comment]
پروکسی اختصاصی دیگه فیلتر نمیشه چون خیلی کم استفاده میکنند
ایده من رایگانه و برای دانشجوها هست 
فقط با یک کلیک میتونن یه فیلترشکن خوب درست کنند
لینک رو ببین
https://github.com/WeAreMahsaAmini/Core/discussions/290
https://hiddify.github.io/setup_proxy.html

**Comment by hiddify-com** [comment]
@IranZop 
ریپازیتوری من رو ببین با یک کلیک سرور روی آژور میاد بالا با همه تنظیمات
https://hiddify.github.io/setup_proxy.html

یعنی یک بچه ای که اصن کامپیوتر هم نمیدونه چیه میتونه راهش بندازه

**Reply by 0xland** [reply]
Feel free to open a pull request.
https://github.com/WeAreMahsaAmini/FreeInternet

**Comment by theStratus** [comment]
خیلی بحث خوبی هست. منم تجربیاتم رو در اختیار میزارم. 

- برای فیلتر شکن، ما از v2ray+websocket استفاده مکینم. خیلی خیلی خوب جواب داده . گرچه وقتی محدود میکنن، گوگل هم زوری باز میشه. برای رفع اون مشکل باید یک سرور تو دیتاسنتر ایران رو واسطه کرد. همون اموزشی که @mona-lisa-1234  توی این [لینک ](https://github.com/WeAreMahsaAmini/Core/discussions/62#discussion-4427309)بهش اشاره کرد یا موارد مشابه.  روی همراه اول مشکلات بشتره گرچه از پورت 443 استفاده کنید به نظر بهتر پیش میره. یک اموزش خوب فارسی و ساده اینه ( [لینک](https://github.com/freeNet4Iran/simple_v2ray_setup) ) اینجا هم پی دی افش هست ([لینک](https://t.me/freeNet4Iran/4)). 
- دیجیتال اوشن هم ارایه دهنده خوبیه و 200 دلار اعتبار اولیه میده (اگر از لینک ریفرال کسی برید)
- هرتزنر هم خوبه، ولی من یک سرور گرفتم از اساس آی پیش فیلتر بود و یکی دیگه گرفتم اوکی بود. کنسول تحت وبش هم مزحرف بود و برای کسی که ن وارده و باید ssh بزنه ممکنه سخت باشه. 
- کسانی که خارج از کشور هستند راحت میتونند از هر جایی سرویس بگیرند. به نظرم دیجیتال اوشن گرچه قیمتش بهترین نیست ولی خیلی اسونه. گرچه با دویست دلار اعتبار اولبه که میده دو ماه اول رایگانه. کنسول تحت وب خوش دستی هم میده. از هرتزنر واقعا بهتره. جوری که نیاز نیست واقعا کسی که بلد نیست ssh بزنه. 
- برای کسانی که داخل ایران هستند هم خوبه که خودشون با پرداخت کریپتوکارنسی، سرور شخصی و امن درست کنند. مثلا Bluevps این سرویس رو میده . یک راهنما اینجا هست . [لینک](https://t.me/freeNet4Iran/28)
- همون طور که @hiddify  عزیز گفتن، بهترین مقباس برای اینکه بلند مدت بشه نگهداریش کرد و روش حساب باز کرد و فیلتر نشه، مقباس کوچیک هست. 20 تا 30 نفر روی هر سرور. ما تجربه 40 روز فیلتر نشدن و سرعت بالا رو داشته ایم. لینک های پابلیک زود فیلتر میشن چون ترافیکشون تابلوعه. 
- تجربه ما هم میگه که openVPN و سرویس های معروف دیگه خیلی زود شناسایی و فیلتر میشوند. v2Ray با websocket و روی http ترافیک رو میبره و واقعا نامحصوصه. میشه obfs یا TLS هم روش زد که یکم TLSکندش میکنه. مخصوصا اگر از CDN (مثلا cloud flare) استفاه بشه. @mona-lisa-1234 @arashmhashemi

**Reply by poorp** [reply]
سلام، v2ray+websocket مشکل DNS داره و سایت ها رو باز نمیکنه و نیاز به تنظیمات خاصی داره.
لطفا اسم سرویس دهنده ها مخصوصا کمتر شناخته شده ها رو هم توی جاهای پابلیک مثل گیت هاب نیارید چون زود بلاک میشن. هر کسی لازم داشته باشه با یه سرچ ساده میتونه پیدا کنه.

**Reply by theStratus** [reply]
ممنونم. فقط روی همراه اول کاربرا مشکل دارند. این مشکل دی ان اس که گفتید رو توضیح میدید یا لینک به یک ایشو که بدونم چی منظورتونه؟

در‌مورد بلاک کردن سرویس ها، خب پنل کاربری خیلی از سرویس ها یا از ایران فیلتر شدن، یا از اونور. اول و اخرش نمیشه شسته رفته بهشون رسید. منظورتون اینه که رنج آی پی رو بلاک میکنند؟ موردی هست که این اتفاق افتاده باشه؟

**Reply by poorp** [reply]
@theStratus 
بله متاسفانه دیدیم که کل سرورهای سرویس دهنده ای به خاطر یک یا چند دوست عزیزی که توی تلگرام یا اینستاگرام برای خودشون کاسبی راه انداختن و اکثرا هم سواد درست درمون ندارن و از همینجاها اطلاعات میگیرن و خودشون به تنهایی از پس یه سرچ ساده هم بر نمیان بلاک شده.
درباره مشکل DNS هم اگر ممکن باشه براتون توی Reddit به من پیام بدید میتونم براتون کانفیگ بفرستم که تست کنید.
آیدی من:
u/ThanksOk4942

**Reply by 0xland** [reply]
There is an "Allow Insecure" option in the iOS client that bypasses the certificate authentication, certificate verification works by sending a request to CA, the CA server may be unreachable during intrAnet times, so enabling that option may help. I don't know the equivalent option for the android client.
